app/api/stripe_authorization.py

```diff
@@ -46,38 +42,40 @@ class StripeAuthorizationList(ResourceList):
     Stripe Authorization List Resource
     """
 
-    def query(self, view_kwargs):
-        query_ = self.session.query(StripeAuthorization)
-        if view_kwargs.get('event_id'):
-            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-            query_ = query_.filter_by(event_id=event.id)
-
-        if view_kwargs.get('event_identifier'):
-            event = safe_query(self, Event, 'identifier', view_kwargs['identifier'], 'event_identifier')
-            query_ = query_.filter_by(event_id=event.id)
-        return query_
-
     view_kwargs = True
-    methods = ['GET', ]
     decorators = (api.has_permission('is_organizer', fetch="event_id", fetch_as="event_id"), )
     schema = StripeAuthorizationSchema
     data_layer = {'session': db.session,
-                  'model': StripeAuthorization,
-                  'methods': {
-                      'query': query
-                  }}
-
+                  'model': StripeAuthorization
+                  }
 
 class StripeAuthorizationDetail(ResourceDetail):
     """
     Stripe Authorization Detail Resource by ID
     """
+    def before_get_object(self, view_kwargs):
+        """
+        method to get id of stripe authorization related to an event
+        :param view_kwargs:
+        :return:
+        """
+        if view_kwargs.get('event_identifier'):
```

Comment:
I don't think that event_query can be used here to get the detail. Correct me if I am wrong :)